### PR TITLE
HOTFIX - invoke agent build with 2h timeout due to flakiness.

### PR DESCRIPTION
**Pull request by m-kostrzewa**

### CODE REVIEW

**Comment by m-kostrzewa** [comment]
related https://github.com/codilime/contrail-windows-ci/pull/84

File: CIScripts/Build/BuildFunctions.ps1 [hunk 1]
```diff
@@ -220,11 +221,13 @@ function Invoke-AgentBuild {
...
-            scons $BuildModeOption controller/src/vnsw/contrail_vrouter_api:sdist | Tee-Object -FilePath $LogsPath/build_api.log
-        }
-    })
+    Invoke-UntilSucceeds {
+        $Job.Step("Building API", {
+            Invoke-NativeCommand -ScriptBlock {
+                scons $BuildModeOption controller/src/vnsw/contrail_vrouter_api:sdist | Tee-Object -FilePath $LogsPath/build_api.log
+            }
+        })
+    } -Duration 60*60*2 # 2 hours
```
Comment (mclapinski) [inline CIScripts/Build/BuildFunctions.ps1]: i'd still prefer nr of iterations

File: CIScripts/Common/Exceptions.ps1 [hunk 2]
```diff
@@ -0,0 +1,4 @@
+class CITimeoutException : System.Exception {
+    CITimeoutException([string] $msg) : base($msg) {}
+    CITimeoutException([string] $msg, [System.Exception] $inner) : base($msg, $inner) {}
+}
```
Comment (mclapinski) [inline CIScripts/Common/Exceptions.ps1]: enter

File: CIScripts/Common/Invoke-UntilSucceeds.ps1 [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    } while (((Get-Date) - $StartTime).Seconds -lt $Duration)
+
+    if ($ReturnVal) {
+        return $ReturnVal
+    } else {
+        $NewException = New-Object -TypeName CITimeoutException("Invoke-UntilSucceeds failed.",
+            $LastException)
+        throw $NewException
+    }
+}
```
Comment (mclapinski) [inline CIScripts/Common/Invoke-UntilSucceeds.ps1]: enter

**Review by mclapinski — APPROVED** [review]

**Pull request closed without merging**